components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/MetadataNode.java

```diff
@@ -15,40 +15,149 @@
 import javax.annotation.Nullable;
 
 import org.sonatype.nexus.common.collect.NestedAttributesMap;
+import org.sonatype.nexus.common.entity.Entity;
+import org.sonatype.nexus.common.entity.EntityId;
 
 import org.joda.time.DateTime;
 
+import static com.google.common.base.Preconditions.checkState;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_ATTRIBUTES;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_BUCKET;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_FORMAT;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_LAST_UPDATED;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_NAME;
+
 /**
- * Wraps an {@code OrientVertex} to provide a simpler API for working with stored component and asset metadata.
+ * Wraps an {@code ODocument} to provide a simpler API for working with stored component and asset metadata.
  *
  * @since 3.0
  */
-public interface MetadataNode
-    extends VertexWrapper
+public abstract class MetadataNode<T>
+    extends Entity
 {
+  private boolean newEntity = true;
+
+  private EntityId bucketId;
+
+  private String name;
+
+  private DateTime lastUpdated;
+
+  private String format;
+
+  private NestedAttributesMap attributes;
+
+  public boolean isNew() {
+    return newEntity;
+  }
+
+  T newEntity(final boolean newEntity) {
+    this.newEntity = newEntity;
+    return self();
+  }
+
+  /**
+   * Gets the bucket this is part of.
+   */
+  @Nullable
+  public EntityId bucketId() {
+    return require(bucketId, P_BUCKET);
+  }
+
+  T bucketId(final EntityId bucketId) {
+    this.bucketId = bucketId;
+    return self();
+  }
+
+  /**
+   * Gets the name.
+   */
+  public String name() {
+    return name;
+  }
+
+  /**
+   * Gets the name or throws a runtime exception if undefined.
+   */
+  public String requireName() {
+    return require(name, P_NAME);
+  }
+
+  /**
+   * Sets the name.
+   */
+  public T name(final String name) {
+    this.name = name;
+    return self();
+  }
+
   /**
    * Gets the last updated date or {@code null} if undefined (the node has never been saved).
```

Comment:
Somewhere in here I think it's worth being explicit that this 'lastUpdatedDate' refers to the properties of the metadata node, rather than any of the other meanings (e.g. it's not the date the package was last revised in a nuget repo, for instance). (I know that's not in the scope of the changes you were making).